### PR TITLE
docs: add nikita-package and nikita-actions plugins

### PR DESCRIPTION
**Pull request by sergkudinov**

### CODE REVIEW

File: docs/website/plugins/nikita-packages/gatsby-node.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+exports.sourceNodes = async ({ actions, createNodeId }) => {
+  const { createNode } = actions
+  // Get all packages
+  const nikitaDir = `${__dirname}/../../../../packages`
+  fs.readdirSync(nikitaDir)
+    .filter(file => fs.statSync(path.join(nikitaDir, file)).isDirectory() && file != 'nikita')
+    .forEach(package => {
+      const packageJsonPath = path.join(nikitaDir, package, 'package.json')
+      if(!fs.existsSync(packageJsonPath)) return
```
Comment (wdavidw) [inline docs/website/plugins/nikita-packages/gatsby-node.js:37]: again

File: docs/website/plugins/nikita-packages/gatsby-node.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    }
+  `)
+}
+
+exports.sourceNodes = async ({ actions, createNodeId }) => {
+  const { createNode } = actions
+  // Get all packages
+  const nikitaDir = `${__dirname}/../../../../packages`
+  fs.readdirSync(nikitaDir)
+    .filter(file => fs.statSync(path.join(nikitaDir, file)).isDirectory() && file != 'nikita')
```
Comment (wdavidw) [inline docs/website/plugins/nikita-packages/gatsby-node.js:34]: oh no, you are blocking the all gatsby generation process

File: docs/website/plugins/nikita-packages/gatsby-node.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
+
+const crypto = require("crypto")
+const path = require("path")
+const fs = require('fs')
```
Comment (wdavidw) [inline docs/website/plugins/nikita-packages/gatsby-node.js:4]: wrong, it shall be `const fs = require('fs').promises

File: docs/website/plugins/nikita-packages/gatsby-node.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      description: String
+      keywords: [String]
+    }
+  `)
+}
+
+exports.sourceNodes = async ({ actions, createNodeId }) => {
+  const { createNode } = actions
+  // Get all packages
+  const nikitaDir = `${__dirname}/../../../../packages`
```
Comment (wdavidw) [inline docs/website/plugins/nikita-packages/gatsby-node.js:32]: shall be obtained from the configuration

File: docs/website/src/templates/package.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+const styles = theme => ({})
+
+const Template = ({
+  data
+}) => {
+  const { page } = data // data.markdownRemark holds our post data
+  return (
+    <Layout page={{...page.frontmatter, slug: page.slug, edit_url: page.edit_url}}>
+      <h1>{page.frontmatter.title}</h1>
+      <p>{page.frontmatter.description}</p>
```
Comment (wdavidw) [inline docs/website/src/templates/package.js:19]: inject the readme

File: docs/website/src/templates/package.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    </Layout>
+  )
+}
+export default withStyles(styles, { withTheme: true })(Template)
```
Comment (wdavidw) [inline docs/website/src/templates/package.js:23]: This is not how style shall be used

**Review by wdavidw — CHANGES_REQUESTED** [review]

File: docs/website/plugins/nikita-packages/gatsby-node.js (hunk 1 above)
Comment (sergkudinov) [reply docs/website/plugins/nikita-packages/gatsby-node.js:37]: ok

File: docs/website/plugins/nikita-packages/gatsby-node.js (hunk 2 above)
Comment (sergkudinov) [reply docs/website/plugins/nikita-packages/gatsby-node.js:34]: not for a long, ok

**Comment by sergkudinov** [comment]
I've put the Mdx type into `readme` property. I also tried into `parent`, but it didn't work on `createResolvers()`.

File: docs/website/gatsby-config.js [hunk 7]
```diff
@@ -91,14 +97,24 @@ module.exports = {
...
+          '**/lib',
+          '**/test',
+          '**/node_modules',
+          '**/src/utils',
+          '**/core/src/plugins',
+          '**/core/src/schedulers',
+          '**/core/src/session',
+        ]
+      },
+    },
```
Comment (wdavidw) [inline docs/website/gatsby-config.js:117]: I am not sure this is the right approach. First, you shall first try to create your own source plugin, second, your globbing expressions are not efficient. No chance for them to optimize the file system discovery.

File: docs/website/plugins/nikita-packages/gatsby-node.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+            },
+            type: "Mdx",
+            firstOnly: false,
+          })
+          return res[0]
+        }
+      }
+    }
+  })
+}
```
Comment (wdavidw) [inline docs/website/plugins/nikita-packages/gatsby-node.js:99]: The readme information are part of your NikitaPackage type, they can potentially contains metadata

File: docs/website/plugins/nikita-packages/gatsby-node.js [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+exports.createPages = ({ actions, graphql }) => {
+  const { createPage, createRedirect } = actions
+  const template = path.resolve(`src/templates/package.js`)
+  return graphql(`
+    {
+      allNikitaPackage {
```
Comment (wdavidw) [inline docs/website/plugins/nikita-packages/gatsby-node.js:106]: packages: allNikitaPackage

**Review by wdavidw — CHANGES_REQUESTED** [review]

File: docs/website/plugins/nikita-packages/gatsby-node.js (hunk 8 above)
Comment (sergkudinov) [reply docs/website/plugins/nikita-packages/gatsby-node.js:99]: so far, there are `title`, `titleHtml`, `tableOfContents` and `body`. To put `frontmatter` of Readme to `NikitaPackage.frontmatter` and keep `body`, an additional request is needed:
```
  createResolvers({
    NikitaPackage: {
      readme: {
        type: "Mdx",
        async resolve(source, args, context, info) {
          const res = await context.nodeModel.runQuery({
            query: {
              filter: {
                slug: {eq: `${source.shortName}/README`}
              },
            },
            type: "Mdx",
            firstOnly: false,
          })
          return res[0]
        }
      },
      frontmatter: {
        async resolve(source, args, context, info) {
          const res = await context.nodeModel.runQuery({
            query: {
              filter: {
                slug: {eq: `${source.shortName}/README`}
              },
            },
            type: "Mdx",
            firstOnly: false,
          })
          return {
            ...res[0] !== undefined ? res[0].frontmatter : null,
            ...source.frontmatter
          }
        }
      }
    }
  })
```
I didn't find any other way.

File: docs/website/plugins/nikita-packages/gatsby-node.js (hunk 9 above)
Comment (sergkudinov) [reply docs/website/plugins/nikita-packages/gatsby-node.js:106]: Using aliases is justified when running multiple queries at the same time. To me, it adds extra syntax complexity in that case. no?

File: docs/website/gatsby-config.js (hunk 7 above)
Comment (sergkudinov) [reply docs/website/gatsby-config.js:117]: I couldn't find a way to negate ignoring the `core/src/actions` folder. Another way is to call the plugin twice:
```
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        path: `${__dirname}/../../packages`,
        name: 'actions',
        ignore: [
          '**/assets',
          '**/env',
          '**/lib',
          '**/test',
          '**/node_modules',
          '**/src/utils',
          '**/core'
        ]
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        path: `${__dirname}/../../packages/core/src/actions`,
        name: 'actions'
      },
    },
```
I will try with an own plugin

**Pull request merged**